lib/gst/player/gstplayer.c

```diff
@@ -112,6 +122,9 @@ struct _GstPlayerPrivate
   GstPlayerState app_state;
   gint buffering;
 
+  GstPlayerMediaInfo  *media_info;  /* current media info */
+  GstPlayerMediaInfo  *media_info_copy; /* last emited media info */
```

Comment:
Why do you store both? Just always store the very latest one, and replace that with a modified copy after you updated it